test/utils/utilities.py

```diff
@@ -19,25 +19,11 @@ def str_to_felt(text: str) -> int:
 def uint(a: int) -> Tuple[int, int]:
     return (a, 0)
 
-
-async def assert_revert(expression, expected_message: Optional[str] = None, expected_code: Optional[StarknetErrorCode] = None):
-    if expected_code is None:
-        expected_code = StarknetErrorCode.TRANSACTION_FAILED
-    try:
-        await expression
-        assert False, "Looks like the transaction didn't revert"
-    except StarkException as err:
-        _, error = err.args
-        assert error['code'] == expected_code, f"assert expected: {expected_code}, got error: {error['code']}"
-        if expected_message is not None:
-            errors_found = [s.removeprefix("Error message: ") for s in error['message'].splitlines() if s.startswith("Error message: ")]
-            assert expected_message in errors_found, f"assert expected: {expected_message}, found errors: {errors_found}"
```

Comment:
Not quite sure I understand your points 1 and 2. `pytest.raises` does the same thanks to the `match` argument. Try changing the match string to something non-sensible and run the test to see it.

As for 3, yeah, that's true. Matter of preference I guess. FWIW, I've been using `pytest.raises` since Cairo 0.6.

Reply:
Hi @milancermak, sorry about the late replay.
Regarding point 1: i bet you can do that with the `match` argument, but that will require you do add `Error message: ` everywhere hurting readability. And that links to point 3, if someday starknet changes the string we need to update all those places

And there is also point 2, raises won't print the real revert reason which is helpful for debugging purposes

Reply:
I still don't follow. You don't need to add `Error message: ` anywhere. The tests in this PR pass. Also no need to adapt if StarkNet changes the prepended string, as the string in `match` is coming from the Argent codebase - it's testing only what Argent throws; [match](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-raises) is a regexp.

If the code inside the `raises` context manager doesn't raise and match the string, pytest will definitely print out the stack trace, it's not difficult to debug at all.

However, as I mentioned in the original post, if you don't like the PR, feel free to close. I submitted it because I though it will make the code base better. Your opinion differs and that's fine, no hard feelings.

Reply:
let me try to explain my point:
By adding  `Error message: ` we make the check more specific. It won't accept any string inside the error message but only revert reasons. (I think we might rename `expected_message` to `revert_reason` to be clear about it) That reduces the number of false positives.

Also, regarding printing the actual revert reason. `error['message']` is usually huge and finding the actual revert reason there is hard to do manually. But our code does it automatically